pandas/io/sas/sas7bdat.py

```diff
@@ -644,8 +641,14 @@ def _process_columnname_subheader(self, offset: int, length: int) -> None:
             )
             col_len = self._read_int(col_name_length, const.column_name_length_length)
 
-            name_str = self.column_names_strings[idx]
-            self.column_names.append(name_str[col_offset : col_offset + col_len])
+            name_raw = self.column_names_raw[idx]
+            cname = name_raw[col_offset : col_offset + col_len]
+            if self.convert_header_text:
```

Comment:
Could you make this pattern into a helper function? It appears it similarly appears 2 more times below.

Reply:
Done.